modin/pandas/indexing.py

```diff
@@ -424,23 +424,21 @@ def _broadcast_item(self, row_lookup, col_lookup, item, to_shape):
         # the target the user is trying to overwrite. This
         if isinstance(item, (pandas.Series, pandas.DataFrame, Series, DataFrame)):
             # convert indices in lookups to names, as Pandas reindex expects them to be so
+            kw = {}
             index_values = self.qc.index[row_lookup]
-            if not all(idx in item.index for idx in index_values):
-                raise ValueError(
-                    "Must have equal len keys and value when setting with "
-                    + "an iterable"
-                )
+            if len(index_values) < len(item.index) or not all(
+                idx in item.index for idx in index_values
+            ):
+                kw["index"] = index_values
```

Comment:
```suggestion
            kw["index"] = index_values
```
don't we want to do this unconditionally? Imagine an `item` having a `[2, 1, 3]` index and `index_values=[1, 2, 3]`, the values don't conform for this if-statement but the `item` still has to be reindexed as the indices aren't equal.

But if we decide to leave this if-statement, I would suggest to use a [`.difference`](https://pandas.pydata.org/docs/reference/api/pandas.Index.difference.html) method here, that should be faster than a python for-loop:
```suggestion
            if len(index_values) < len(item.index) or len(index_values.difference(item.index) != 0:
                kw["index"] = index_values
```

Reply:
We have such a test case and it works (`# len(key) == df_value.columns; different index`).

Reply:
The reason the test you mentioning passes is that the data being assigned is constant, and so doesn't depend at all from the order of assigning:
```python
(Pdb) df_value
     col1  col2
255     1     2
254     1     2
253     1     2
252     1     2
251     1     2
..    ...   ...
4       1     2
3       1     2
2       1     2
1       1     2
0       1     2

[256 rows x 2 columns]
```

Once you change the value to be various from index-to-index, the test fails
```diff
index 16381131..9f240d38 100644
--- a/modin/pandas/test/dataframe/test_indexing.py
+++ b/modin/pandas/test/dataframe/test_indexing.py
@@ -409,7 +409,8 @@ def test_loc_4456():
     key = ["col1", "col2"]

     # len(key) == df_value.columns; different columns
-    value = [[1, 2]] * pandas_df.shape[0]
+    # value = [[1, 2]] * pandas_df.shape[0]
+    value = [[i, i + 1] for i in range(pandas_df.shape[0])]
     df_value = pandas.DataFrame(value, columns=["value_col1", "value_col2"])
     eval_loc(modin_df, pandas_df, df_value, (slice(None), key))
```

Reply:
@dchigarev fixed + updated on random data

Reply:
@dchigarev ready for review